pubsub/client.go

```diff
@@ -482,11 +286,77 @@ func (c *PubSubClient) GetSchema(schemaID string) (*eventbusv1.SchemaInfo, error
 	return resp, nil
 }
 
+func (c *Client) Publish(ctx context.Context, records []opencdc.Record) error {
+	var events []*eventbusv1.ProducerEvent
+	// TODO - refactor the way we access the topic
+	topic, err := c.GetTopic(c.topicNames[0])
+	if err != nil {
+		return fmt.Errorf("error on publish, cannot retrieve topic %s : %s", c.topicNames[0], err)
+	}
+
+	sdk.Logger(ctx).Info().
+		Str("topic", topic.TopicName).
+		Msg("Started publishing event")
+
+	codec, err := c.fetchCodec(ctx, topic.SchemaId)
+	if err != nil {
+		return err
+	}
+
+	for _, r := range records {
+		data := r.Payload.After
+
+		var dataMap map[string]interface{}
```

Comment:
You don't need to do this, `r.Payload.After` needs to be `opencdc.StructuredData` which is essentially `map[string]any` and already serialized to go types. Doing this JSON unmarshal from bytes all bad things can happen.